cli/src/artifacts.ts

```diff
@@ -58,15 +62,15 @@ export default class Artifacts {
     if (!JSON.isObject(json)) {
       throw new TypeError("expected object, found " + (json === null ? "null" : typeof json));
     }
-    let active = json.active;
+    let { active } = json as ArtifactsOutput;
     if (typeof active !== 'string' && active !== null) {
       throw new TypeError("json.active is not a string or null");
     }
-    let jsonTargets = json.targets;
+    let { targets: jsonTargets } = json as any;
```

Comment:
Again, would `unknown` work here?

I'm a big fan of destructuring generally, but when we have to add the extra cast, I'm not sure how much this refactor to use destructuring is really buying us. The cast feels kind of noisy and irrelevant to the reader. Sam with the `let { active } = ...` line above.